File: jsimple-json/src/main/java/jsimple/json/readerwriter/JsonLongProperty.java
package jsimple.json.readerwriter;

/**
 * @author Bret Johnson
 * @since 5/22/13 9:08 PM
 */
public class JsonLongProperty extends JsonProperty {
    public JsonLongProperty(String name, int id) {
        super(name, id);
    }

    // TODO: Automatically convert int to long
    public long readValue(JsonObjectReader objectReader) {
        Object value = objectReader.readPropertyValue();
        if (value instanceof Integer)
            return (long) (int) (Integer) value;
        else return (long) (Long) value;
    }

    @Override public Object readValueUntyped(JsonObjectReader objectReader) {
        return objectReader.readPropertyValue();
    }

    public void write(JsonObjectWriter objectWriter, long value) {
        objectWriter.writeProperty(this, value);
    }
}

File: jsimple-json/c#/jsimple/json/readerwriter/JsonLongProperty.cs
namespace jsimple.json.readerwriter
{

	/// <summary>
	/// @author Bret Johnson
	/// @since 5/22/13 9:08 PM
	/// </summary>
	public class JsonLongProperty : JsonProperty
	{
		public JsonLongProperty(string name, int id) : base(name, id)
		{
		}

		// TODO: Automatically convert int to long
		public virtual long readValue(JsonObjectReader objectReader)
		{
			return (long)(long?) objectReader.readPropertyValue();
		}

		public override object readValueUntyped(JsonObjectReader objectReader)
		{
			return objectReader.readPropertyValue();
		}

		public virtual void write(JsonObjectWriter objectWriter, long value)
		{
			objectWriter.writeProperty(this, value);
		}
	}

}